src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SimpleConverterTypeComparer.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    internal sealed class SimpleConverterTypeComparer : IEqualityComparer<Type>
```

Comment:
What is this at all? `Type` can just be compared by reference when used in dictionary.